src/main/scala/org/xalgorithms/apps/KafkaMongoSparkStreamingApplication.scala

```diff
@@ -0,0 +1,75 @@
+package org.xalgorithms.apps
+
+import java.lang.management.ManagementFactory
+
+import com.mongodb.client.MongoCollection
+import com.mongodb.spark.MongoConnector
+import com.mongodb.spark.config.WriteConfig
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.streaming.{Seconds, StreamingContext}
+import org.apache.spark.streaming.dstream.DStream
+import org.bson.Document
+import scala.collection.JavaConverters._
+
+
```

Comment:
You should do this subclassing properly. It seems to me like the KafkaMongoSink is the only thing that changes in the with_context implementation that was cloned from the base class. Find a way to parameterized the KafkaSink in the base class using KafkaMongoSink in *this* class and KafkaSink in the base class.